c2rust-analyze/tests/common/mod.rs

```diff
@@ -13,6 +13,7 @@ use clap::Parser;
 #[derive(Default)]
 pub struct Analyze {
     path: PathBuf,
+    dont_catch_panic: bool,
```

Comment:
Instead of adding this here, I'd add it as a directive arg like this (before `AnalyzeArgs::env`):

```rust
    /// Don't catch any panics during analysis of functions.
    ///
    /// The default behavior is to catch panics,
    /// try to recover as much as possible,
    /// and analyze as many functions as possible.
    /// Setting this will make panics fatal instead.
    #[arg(long)]
    dont_catch_panic: bool,
```

and then add `//! --dont-catch-panic` to all of the current tests, `analyze` and `filecheck` ones, as that was their previous behavior and we want to keep testing the same behavior.  When we add a new test or want to change an existing test to test behavior post-panicking, then we can remove the `//! --dont-catch-panic` from that file.

Reply:
I'm going to turn on "don't catch panic" mode by default in all tests and add a `//! --catch-panic` option to opt out instead.  Catching panics by default covered up a significant regression in #919, which I originally developed on top of this branch.

Reply:
That makes sense.  I was wondering, too, if the default should be panicking, too.  nit: I'd call it `--catch-panics`, though.